crates/typst-library/src/math/lr.rs

```diff
@@ -107,18 +127,131 @@ pub fn abs(
 #[func]
 pub fn norm(
     /// The size of the brackets, relative to the height of the wrapped content.
+    ///
+    /// Default: The current value of [`lr.size`]($math.lr.size).
     #[named]
     size: Option<Rel<Length>>,
     /// The expression to take the norm of.
     body: Content,
 ) -> Content {
-    delimited(body, '‖', '‖', size)
+    delimited(body, '‖'.into(), '‖'.into(), size)
+}
+
+/// Gets the Left/Right wrapper function corresponding to a left delimiter, if
+/// any.
+pub fn get_lr_wrapper_func(left: &str) -> Option<Func> {
+    match left {
+        "⌈" => Some(ceil::func()),
+        "⌊" => Some(floor::func()),
+        l => FUNCS.get(l).map(Func::from),
+    }
+}
+
+/// The delimiter pairings supported for use as callable symbols.
+const DELIMS: &[(&str, &str)] = &[
+    ("(", ")"),
+    ("⟮", "⟯"),
+    ("⦇", "⦈"),
+    ("⦅", "⦆"),
+    ("⦓", "⦔"),
+    ("⦕", "⦖"),
+    ("{", "}"),
+    ("⦃", "⦄"),
+    ("[", "]"),
+    ("⦍", "⦐"),
+    ("⦏", "⦎"),
+    ("⟦", "⟧"),
+    ("⦋", "⦌"),
+    ("❲", "❳"),
+    ("⟬", "⟭"),
+    ("⦗", "⦘"),
+    ("⟅", "⟆"),
+    ("⎰", "⎱"),
+    ("⎱", "⎰"),
+    ("⧘", "⧙"),
+    ("⧚", "⧛"),
+    ("⟨", "⟩"),
+    ("⧼", "⧽"),
+    ("⦑", "⦒"),
+    ("⦉", "⦊"),
+    ("⟪", "⟫"),
+    ("⌜", "⌝"),
+    ("⌞", "⌟"),
+];
+
+/// Lazily created left/right wrapper functions.
+static FUNCS: LazyLock<HashMap<&'static str, NativeFuncData>> = LazyLock::new(|| {
+    let bump = Box::leak(Box::new(Bump::new()));
+    DELIMS
+        .iter()
+        .copied()
+        .map(|(l, r)| (l, create_lr_func_data(l.into(), r.into(), bump)))
+        .collect()
+});
+
+/// Creates metadata for an L/R wrapper function.
+fn create_lr_func_data(
+    left: EcoString,
+    right: EcoString,
+    bump: &'static Bump,
+) -> NativeFuncData {
+    let title = bump.alloc(format!("{left}{right} Left/Right"));
```

Comment:
This could use https://docs.rs/bumpalo/latest/bumpalo/macro.format.html + https://docs.rs/bumpalo/latest/bumpalo/collections/string/struct.String.html#method.into_bump_str to avoid the intermediate allocation.